sink_service/source/main.c

```diff
@@ -60,17 +60,109 @@ static bool get_service_name(char service_name[MAX_SIZE_SERVICE_NAME], unsigned
     return true;
 }
 
+/**
+ * \brief   Attempts to acquire an environment variable
+ * \param   dest
+ *          The destination buffer where to store the read value
+ * \param   name
+ *          The name of the environment variable to acquire
+ * \return  A pointer to the character buffer
+ */
+static bool get_env_var(char * dest, char * name)
+{
+    char * ptr = NULL;
+    int ret;
+
+    memset(dest, '\0', sizeof(*dest) * MAX_SIZE_PARAMETER);
+    if ((ptr = getenv(name)) != NULL)
+    {
+        ret = snprintf(dest, sizeof(*dest) * (MAX_SIZE_PARAMETER - 1), "%s", ptr);
+        if (ret < 0)
+        {
+            LOGW("Input var (%s) too long for destination\n", name);
+            return false;
+        }
+        LOGI("%s: %s\n", name, dest);
+        return true;
+    }
+
+    return false;
+}
+
+/**
+ * \brief   Obtains environment parameters to control sink settings
+ * \param   bitrate
+ *          Pointer where to store bitrate value (if any)
+ * \param   port_name
+ *          Pointer where to store port_name value (if any)
+ * \param   sink_id
+ *          Pointer where to store sink_id value (if any)
+ * \param   max_poll_fail_duration
+ *          Pointer where to store max_poll_fail_duration value (if any)
+ * \return  Number of acquired parameters
+ */
+static unsigned int get_env_parameters(unsigned long * bitrate,
+                                       char * port_name,
+                                       unsigned int * sink_id,
+                                       unsigned int * max_poll_fail_duration)
+{
+    char dest[MAX_SIZE_PARAMETER];
+    unsigned int acquired = 0;
+
+    if (get_env_var(dest, "WM_GW_SINK_ID"))
+    {
+        *sink_id = strtoul(dest, NULL, 0);
+        acquired += 1;
+    }
+
+    if (get_env_var(dest, "WM_GW_SINK_BITRATE"))
+    {
+        *bitrate = strtoul(dest, NULL, 0);
+        acquired += 1;
+    }
+
+    if (get_env_var(dest, "WM_GW_SINK_MAX_POLL_FAIL_DURATION"))
+    {
+        *max_poll_fail_duration = strtoul(dest, NULL, 0);
+        acquired += 1;
+    }
+
+    if (get_env_var(dest, "WM_GW_SINK_UART_PORT"))
+    {
+        int ret;
+
+        memset(dest, '\0', sizeof(*port_name) * MAX_SIZE_PARAMETER);
+        ret = snprintf(port_name, sizeof(*port_name) * (MAX_SIZE_PARAMETER - 1), "%s", dest);
+        if (ret < 0)
+        {
+            LOGW(
```

Comment:
Should be an error no? And maybe exit the program. User specified a global var that will not be used

Reply:
I thought about exiting in error or simply letting that be handled later on by the code. Would you prefer to exit in error at this point?